src/Inciter/Discretization.cpp

```diff
@@ -144,12 +133,40 @@ Discretization::Discretization(
     m_fct[ thisIndex ].insert( m_nchare, m_gid.size(), nprop,
                                m_nodeCommMap, m_bid, m_lid, m_inpoel );
 
-  #ifdef EXAM2M
-  // Establish bridge to ExaM2M mesh-transfer lib
-  tk::Fields u;
-  m_m2m = CProxy_CharmMesh:ckNew( m_inpoel, m_coord, m_nchare, u,
-                                  CkArrayOptions().bindTo(thisProxy) );
+  // Register mesh with mesh-transfer lib
+  #ifdef HAS_EXAM2M
+  if (thisIndex == 0) {
+    exam2m::addMesh( thisProxy, m_nchare,
+      CkCallback( CkIndex_Discretization::transferInit(), thisProxy ) );
+    std::cout << "Disc: " << m_meshid << " m2m::addMesh()\n";
+  }
+  #else
+  transferInit();
   #endif
+}
+
+void
+Discretization::transferInit()
+// *****************************************************************************
+// Our mesh has been registered with the mesh-to-mesh transfer library (if
+// coupled to other solver)
+// *****************************************************************************
+{
+  // Lambda to decide if a node is not counted by this chare. If a node is
+  // found in the node communication map and is associated to a lower chare id
+  // than thisIndex, it is counted by another chare (and not thisIndex), hence
+  // a "slave" (for the purpose of this count).
+  auto slave = [ this ]( std::size_t p ) {
```

Comment:
Interesting. Does this mean we won't be testing for duplicate collisions in ExaM2M? It's still probably a feature we should add to ExaM2M itself as well maybe? But if it works here that's good.

Reply:
No, this is only used to sum the total number of mesh points without multi-counting those on the chare-boundary. We still pass the mesh connectivity and the node coordinates to ExaM2M in which the same physical points on chare-boundaries will show up multiple times bot on different chares.